lib/formats/html/html.flow

```diff
@@ -490,67 +494,91 @@ urlEncode(s){
 	|> r("™", "%E2%84%A2")
 }
 
-urlDecode(s) {
+urlDecode(s : string) -> string {
 	urlDecode0("", s)
 }
 
-urlDecode0(acc, s) {
+urlDecode0(acc : string, s : string) -> string {
 	percent = strIndexOf(s, "%");
 	if (percent == -1) {
 		acc + s;
 	} else {
 		left = strLeft(s, percent);
 		right = strRight(s, percent);
-		percentPair = extractPercentHex(right);
-		hexcode = percentPair.first;
-		c = if (hexcode < 0xC0) {
-			percentPair
-		} else {
-			// OK, it must be UTF-8 encoded
-			if (0xC0 <= hexcode && hexcode <= 0xDF) {
-				// 2 bytes
-				secondByte = extractPercentHex(percentPair.second);
-				if (secondByte.first == -1) {
-					// OK, it is some crap, because we expect a second percent here
-					percentPair
-				} else {
-					Pair((hexcode - 192) * 64 + (secondByte.first - 128), secondByte.second)
-				}
-			} else if (hexcode <= 0xEF) { // 3 bytes
-				secondByte = extractPercentHex(percentPair.second);
-				thirdByte = extractPercentHex(secondByte.second);
-				if (secondByte.first == -1 || thirdByte.first == -1) {
-					// OK, it is some crap, because we expect a second and third percent here
-					percentPair
-				} else {
-					Pair((hexcode - 224) * 4096
-						+ (secondByte.first - 128) * 64
-						+ (thirdByte.first - 128),
-						thirdByte.second
-					)
-				}
-			} else if (hexcode <= 0xF7) { // 4 bytes
-				secondByte = extractPercentHex(percentPair.second);
-				thirdByte = extractPercentHex(secondByte.second);
-				fourthByte = extractPercentHex(secondByte.second);
```

Comment:
btw shouldn't it be fourthByte = extractPercentHex(thirdByte.second); ? this implies a change in derived code too

Reply:
sure, this is a bug at initial code, fixed, thx